pkg/controller/operators/catalog/operator.go

```diff
@@ -886,10 +886,12 @@ func (o *Operator) syncCatalogSources(obj interface{}) (syncError error) {
 func (o *Operator) isFailForwardEnabled(namespace string) (bool, error) {
 	ogs, err := o.lister.OperatorsV1().OperatorGroupLister().OperatorGroups(namespace).List(labels.Everything())
 	if err != nil {
-		return false, fmt.Errorf("couldn't list operatorGroups, assuming default upgradeStrategy")
+		// Couldn't list operatorGroups, assuming default upgradeStrategy
+		// so existing behavior is observed for failed CSVs.
+		return false, nil
 	}
 	if len(ogs) != 1 {
-		return false, fmt.Errorf("found %d operatorGroups, expected 1, assuming default upgradeStrategy", len(ogs))
+		return false, fmt.Errorf("found %d operatorGroups in namespace %s, expected 1", len(ogs), namespace)
```

Comment:
Does error-ing out here get reflected back to the Subscription's status?

Reply:
We could do something like [this](https://github.com/operator-framework/operator-lifecycle-manager/blob/8abe078086bba81e2765a244b5a16fe1196f0f7c/pkg/controller/operators/catalog/operator.go#L972)

Reply:
Yeah that seems reasonable to me? Maybe handle that at the call site though (e.g. o.syncResolvingNamespaces)?

Reply:
o.syncResolvingNamespaces is definitely the correct place to do this, but:
- We'll need to introduce a new [SubscriptionConditionType](https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/subscription_types.go#L91)
- Other errors in the function are not reflected in the status.
I feel like we ought to address this holistically, possibly in a follow up PR, thoughts?

Reply:
Talked in slack: this is fine with me as a follow-up.